### PR TITLE
opentracing: remove legacy open tracing extension

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: medium
Testing: n/a
Docs Changes:
Release Notes: inline
Fixes https://github.com/envoyproxy/envoy/issues/27401
Fixes https://github.com/envoyproxy/envoy/issues/34321

### CODE REVIEW

**Review by htuch — APPROVED** [review]

File: api/envoy/config/trace/v3/dynamic_ot.proto [hunk 1]
```diff
@@ -20,10 +20,9 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
 // [#protodoc-title: Dynamically loadable OpenTracing tracer]
 
-// DynamicOtConfig is used to dynamically load a tracer from a shared library
+// DynamicOtConfig was used to dynamically load a tracer from a shared library
```
[the remark this answers is not in the document]
Comment (wbpcode) [reply api/envoy/config/trace/v3/dynamic_ot.proto:23]: I have provided file level hiding. But by the way, why we don't remove this directly?

[the remark this answers is not in the document]
Comment (wbpcode) [reply api/envoy/config/trace/v3/dynamic_ot.proto:23]: But now any one use the API will only get an error? (Although I have no strong point to that.

[the remark this answers is not in the document]
Comment (wbpcode) [reply api/envoy/config/trace/v3/dynamic_ot.proto:23]: > i dont think its hidden currently (or in this PR)

Yeah. I mean we can hide this file completely. This is a optional suggestion to @alyssawilk

**Review by phlax — APPROVED** [review]
we can also remove the opentracing dep setup from bazel/repos - but lets do that in a follow up

lgtm, thanks @alyssawilk

**Pull request merged**